### PR TITLE
Include the partial boolean lib in builds, and allow removing constraints from a DependencyHelper

### PR DESCRIPTION
**Pull request by AlexIIL**
Needed for https://github.com/QuiltMC/quilt-loader/pull/11

### CODE REVIEW

**Comment by AlexIIL** [comment]
I've had to exclude a number of files since they had compile errors :/

**Comment by zml2008** [comment]
were the partial boolean compile errors a Java version thing? or just that part of the lib straight up didn't work

**Comment by AlexIIL** [comment]
A number of files are referenced (E.G. `org.sat4j.tools.DimacsStringSolver`) which don't seem to be present in the repository?

**Comment by zml2008** [comment]
interesting -- that appears to be available [upstream](https://gitlab.ow2.org/sat4j/sat4j/-/blob/master/org.sat4j.core/src/main/java/org/sat4j/tools/DimacsStringSolver.java).

I wonder if Fabric stripped out unused class files when originally forking?

Perhaps it makes sense to switch this over to a [gitpatcher](https://github.com/MinecraftForge/gitpatcher)-based system instead, to keep better track of changes?

**Comment by zml2008** [comment]
<https://github.com/QuiltMC/quilt-loader-sat4j/commit/76e6c904b5cba2296d157972d72965d96c767826> ah, here -- the files were removed

**Review by TheGlitch76 — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -11,17 +11,20 @@ plugins {
...
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
 version = '2.3.5'
 
 def ENV = System.getenv()
 if (ENV.BUILD_NUMBER) {
     version = version + "." + "${System.getenv().BUILD_NUMBER}"
 }
+// Temp for alexiil's fork
+version += "-alexiil.2"
```
Comment (TheGlitch76) [inline build.gradle]:
```suggestion
```

**Comment by AlexIIL** [comment]
Sorry for the delay in responding.

I think re-adding those files is probably a good idea, as we don't really want to diverge too much from upstream (and minifying the resulting jar is probably not a very high priority). That being said, this does actually contain one change: Allowing constraints to be removed from a `DependancyHelper` object. I should probably rename those methods to be `quilt_*` just to make it much easier to tell the difference.

I'm not sure if it's worth it to use gitpatcher for *just* that?

**Comment by TheGlitch76** [comment]
Realistically, the chances of us needing to change this again are very low. I'm not sure I see any benefit to spending a significant amount of time trying to migrate this repo.

**Comment by zml2008** [comment]
Yeah -- I think it's relatively low-priority, the main concerns should be ease of upstreaming and functionality, and in the short term only #2 matters, #1 can happen once quilt is more stable

**Comment by AlexIIL** [comment]
What version should I use for this? Currently its set to 2.3.5, should we just use build numbers for quilt builds (following sat4j versions) or should we bump manually to follow sem-var a bit better? (I don't mind either way, just since we need *a* version number to depend on :p )

**Comment by TheGlitch76** [comment]
I think adding a build number is the way to go for this, considering we are specifically basing off of a specific version of sat4j and I'd bet this won't be updated again in years.

**Comment by AlexIIL** [comment]
so `2.3.5.1`?

**Comment by TheGlitch76** [comment]
Yeah, that's fine IMO

**Comment by AlexIIL** [comment]
Ok, done.

**Comment by TheGlitch76** [comment]
> I don't see anything glaringly wrong

- i509

Going to go ahead and merge this.

**Pull request merged**